gensim/downloader.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import absolute_import
+import argparse
+import json
+import os
+import tarfile
+import logging
+import sys
+import errno
+import hashlib
+try:
+    import urllib.request as urllib
+except ImportError:
+    import urllib
+
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+user_dir = os.path.expanduser('~')
+base_dir = os.path.join(user_dir, 'gensim-data')
+data_log_file_path = os.path.join(base_dir, 'data.json')
+logger = logging.getLogger('gensim.api')
+
+
+def get_data_list():
+    """Function getting the list of all datasets/models.
+
+    Returns:
+        list: returns a list of datasets/models avalible for installation.
+    """
+    url = "https://raw.githubusercontent.com/RaRe-Technologies/gensim-data/master/list_with_filename.json"
+    response = urlopen(url)
+    data = response.read().decode("utf-8")
+    data = json.loads(data)
+    data_names = []
+    corpora = data['corpus']
+    models = data['model']
+    for corpus in corpora:
+        data_names.append(corpus)
+    for model in models:
+        data_names.append(model)
+    return data_names
+
+
+def get_data_name(data_):
+    """Returns a name for the dataset/model as to download a dataset/model user can alternate names too.
+
+    Args:
+        data_(string): Name of the corpus/model.
+
+    Returns:
+        data_: returns the name for dataset/model
+    """
+    url = "https://raw.githubusercontent.com/RaRe-Technologies/gensim-data/master/alternate_names.json"
+    response = urlopen(url)
+    alternate_names_json = response.read().decode("utf-8")
+    alternate_names_json = json.loads(alternate_names_json)
+    data_names = get_data_list()
+    for data_name in data_names:
+        alternate_data_names = alternate_names_json[data_name]
+        if data_ in alternate_data_names:
+            return data_name
+
+
+def initialize_data_log_file():
+    """Function for initializing the log file. Creates a json object
+    for each corpus/model and stores in the log file. For eg: {"name": "text8", "status" : "None"}
+    """
+    data = info()
+    corpora = data['corpus']
+    models = data['model']
+    json_list = []
+    for corpus in corpora:
+        json_object = {"name": corpus, "status": "None"}
```

Comment:
Why do we need to store anything with "None" status if we check file with data/model from data repo each time?

Reply:
This is for initialising the json log file that stores the status if the model has downloaded or installed on the users computer.